code/modules/power/tesla/energy_ball.dm

```diff
@@ -30,21 +36,31 @@
 	miniball = is_miniball
 	. = ..()
 	if(!is_miniball)
-		set_light(10, 7, "#EEEEFF")
+		set_light_on(TRUE)
+	else if(energy < TESLA_DEFAULT_POWER)
+		transform *= pick(0.3, 0.4, 0.5, 0.6, 0.7)
 
 /obj/singularity/energy_ball/supermatter
 	name = "hypercharged supermatter energy ball"
 	desc = "The supermatter energy ball hovers ominously, a radiant orb of sheer power. Its brilliance is blinding, casting an intense glow that illuminates the surrounding area. The air crackles with the electric energy it exudes. The sheer intensity of its presence instills a sense of caution, reminding you of the untamed force contained within. Wisps of energy escape its surface, dissipating into the atmosphere with a sizzling sound. Sparks of energy occasionally arc between the crystal and the energy ball, crackling with a captivating yet dangerous allure."
 	icon_state = "smenergy_ball"
 	energy = 10000
 	max_balls = 20
+	dissipate = 0
 	zap_range = 7
-	hypercharged = TRUE
+	zap_flags = TESLA_DEFAULT_FLAGS | TESLA_MOB_GIB
```

Comment:
you forgot the ALLOW_TESLA_DUPLICATE flag

Reply:
Intended change, forgot to add it to the changelog, my bad.